cli/cmd/encore/daemon/daemon.go

```diff
@@ -36,11 +36,15 @@ import (
 	"encr.dev/cli/daemon/sqldb/external"
 	"encr.dev/cli/internal/xos"
 	"encr.dev/internal/conf"
+	"encr.dev/pkg/eerror"
+	"encr.dev/pkg/watcher"
 	daemonpb "encr.dev/proto/encore/daemon"
 )
 
 // Main runs the daemon.
 func Main() {
+	watcher.BumpRLimitSoftToHardLimit()
```

Comment:
Go bumps the hard limits, but not the soft limits, so I've added this as a satefy net, even though the rest of the PR means we don't need it in the cases we've identified so far